tests/unit/charms/tls_certificates_interface/v2/test_tls_certificates_v2_provides.py

```diff
@@ -414,6 +612,48 @@ def test_given_identical_csr_in_relation_data_when_set_relation_certificate_then
         loaded_relation_data = _load_relation_data(dict(provider_relation_data))
         self.assertEqual(expected_relation_data, loaded_relation_data)
 
+    def test_given_identical_csr_in_relation_data_when_set_relation_certificate_and_unit_not_leader_then_certificate_is_not_replaced_in_relation_data(  # noqa: E501
```

Comment:
Like above, cases look independent. is the cert not replaced because the csr is identical or because unit is not leader?

Reply:
Likewise, I would like to remove this test, unit is not leader is main reason independent from other conditions.

Reply:
Done